source/normalize-arguments.js

```diff
@@ -15,7 +16,9 @@ module.exports = (url, options, defaults) => {
 		throw new TypeError('Parameter `url` is not an option. Use got(url, options)');
 	}
 
-	if (!is.string(url) && !is.object(url)) {
+	if (options.baseUrl && (is.string(url) || is(url) === 'URL')) {
+		url = urlToOptions(new URLGlobal(url, options.baseUrl));
+	} else if (!is.string(url) && !is.object(url)) {
 		throw new TypeError(`Parameter \`url\` must be a string or object, not ${is(url)}`);
```

Comment:
I think this should be moved out of the if/else into a standalone `if` above:

```js
if (!is.string(url) && !is.object(url)) {
	throw new TypeError(`Parameter \`url\` must be a string or object, not ${is(url)}`);
}

if (options.baseUrl && (is.string(url) || is(url) === 'URL')) {
...
```

Reply:
Yep. Doing right now.